st2actions/st2actions/runners/ssh/paramiko_ssh.py

```diff
@@ -349,6 +349,7 @@ def run(self, cmd, timeout=None, quote=False):
         chan = transport.open_session()
 
         start_time = time.time()
+        chan.get_pty()
```

Comment:
The error in the failure is interesting. 

```invoke-rc.d: -----------------------------------------------------
invoke-rc.d: WARNING: 'invoke-rc.d puppet start' called
invoke-rc.d: during shutdown sequence.
invoke-rc.d: enabling safe mode: initscript policy layer disabled
invoke-rc.d: -----------------------------------------------------```

This usually means that the server is in a different run-level, so when `apt-get` attempts to start the daemon, it bombs out. 

In the `/etc/init.d/puppet` init script...

```
# Default-Start:     2 3 4 5
# Default-Stop:      0 1 6
```

So, the server is in runlevel 0, 1, or 6 here, as this is how it would manifest. So, I'm curious about what else is happening around the workflow. Is the server still starting up, and not fully in runlevel 2? Is something else shutting down the server?

Does this provide any hints?